### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by sbesson**
Upgrade the IDR utility scripts to be Python 3 compatible and fix the study parser utility with OMERO 5.6.

- 993d8b8 was originally generated by running `futurize -w -n .`
- the other commits were tested while updating the annotations on `prod80`. The outcome of these changes can be tested on a local server by running

  ```
   $ git clone git://github.com/sbesson/idr-utils -b python3
   $ /opt/omero/server/venv3/bin/omero login public@localhost -w public
   $ find /uod/idr/metadata/ -type f -name '*study.txt' | sort | xargs /opt/omero/server/venv3/bin/python pyidr/study_parser.py --check -q
  ```

The `stats.py` and other scripts should be minimally functional but have not been tested yet and might need additional tweaks outside the scope of this PR

### CODE REVIEW

File: pyidr/file_pattern.py [hunk 1]
```diff
@@ -57,10 +61,10 @@ def expand_range(r):
             raise ValueError("inverted range: %s" % r)
         step = int(step)
         if len(start_str) != len(stop_str):
-            return map(str, range(start, stop, step))
+            return list(map(str, list(range(start, stop, step))))
```
Comment (mtbc) [inline pyidr/file_pattern.py]: That inner `list` really is necessary? Or could even get rid of both with a list comprehension.

File: scripts/celery/get_failed.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
             fail[taskid] = r
 
 print('SUMMARY')
-for k, v in st.items():
+for k, v in list(st.items()):
```
Comment (mtbc) [inline scripts/celery/get_failed.py]: Again, these `list` wrappings are really necessary?

**Comment by dominikl** [comment]
Studyparser works with python 3. Changes look good. But I don't know about the explicite `list` calls either.

File: scripts/celery/get_failed.py (hunk 2 above)
Comment (sbesson) [reply scripts/celery/get_failed.py]: See 
f400c6f

File: pyidr/file_pattern.py (hunk 1 above)
Comment (sbesson) [reply pyidr/file_pattern.py]: See f400c6f

**Comment by sbesson** [comment]
@dominikl @mtbc fixes pushed for the `list` comprehension although I haven't tested this at all as I was primarily interested in the study annotation utility. Vetoes to merging and fixing issues that arise as follow-ups?

**Comment by mtbc** [comment]
Go for it. :+1:

**Comment by dominikl** [comment]
👍

**Pull request merged**